README.md

```diff
@@ -1,2 +1,116 @@
 # CacheAdvance
+
+[![CI Status](https://travis-ci.com/dfed/CacheAdvance.svg?branch=master)](https://travis-ci.com/dfed/CacheAdvance)
+[![Swift Package Manager compatible](https://img.shields.io/badge/SPM-compatible-4BC51D.svg?style=flat)](https://github.com/apple/swift-package-manager)
+[![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat)](https://github.com/Carthage/Carthage)
+[![Version](https://img.shields.io/cocoapods/v/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+[![License](https://img.shields.io/cocoapods/l/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+[![Platform](https://img.shields.io/cocoapods/p/CacheAdvance.svg)](https://cocoapods.org/pods/CacheAdvance)
+
 A cache that enables the performant persistence of individual messages to disk.
+
+## Usage
+
+### Basic Initialization
+
+```swift
+let myCache = try CacheAdvance<MyMessageType>(
+    file: FileManager.default.temporaryDirectory.appendingPathComponent("MyCache"),
+    maximumBytes: 5000,
+    shouldOverwriteOldMessages: false)
+```
+To begin caching messages, you need to create a CacheAdvance instance with:
+
+* A file URL – this URL must represent a file that has already been created. You can create a file by using `FileManager`'s [createFile(atPath:contents:attributes)](https://developer.apple.com/documentation/foundation/filemanager/1410695-createfile) API.
+* A maximum number of bytes on disk the cache can consume.
+* Whether the cache should overwrite old messages. If you need to preserve every message, set this value to `false`. If you care only about preserving recent messages, set this value to `true`.
+
+### Appending messages to disk
+
+```swift
+try myCache.append(message: aMessageInstance)
+```
+
+By the time the above method exits, the message will have been persisted to disk. A CacheAdvance keeps no in-memory buffer. Appending a new message is cheap, as a CacheAdvance needs to encode and persist only the new message and associated metadata.
+
+A CacheAdvance instance that does not overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if appending a message would exceed the cache's `maximumBytes`. A CacheAdvance instance that does overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if the message would require more than `maximumBytes` to store even after evicting all older messages from the cache.
+
+To ensure that caches can be read from 32bit devices, messages should not be larger than 2GB in size.
```

Comment:
Do we assert for this in code? Or maybe we should throw an exception?

Also are we sure this restriction is real? I thought a 32 bit device still has the concept of `UInt64` but that it was less efficient.

Reply:
We don't assert or throw an exception. Starting with documentation. I can make a separate to enforce this.

32-bit devices do have the concept of `UInt64`, but `Int` will be an `Int32` on a 32-bit device, which means that when we cast the message length to `Int` in order to `readDataUp(toLength:)`, we'll overflow on anything larger than ~2GB.

Reply:
Added a `throw`. See https://github.com/dfed/CacheAdvance/pull/22#discussion_r362278580.

That said, we may still have an issue on 32bit devices, since I have no idea what happens when someone tries to encode a Data blob longer than `Int32`. I'm hoping that trying to encode a message longer than `Int32` on a 32bit device `throws`, but I don't have a good way to check.

Reply:
> I'm hoping that trying to encode a message longer than `Int32` on a 32bit device throws, but I don't have a good way to check.

I see what you’re saying. Hopefully it `throws`...